cmd/compose/compose.go

```diff
@@ -120,22 +120,24 @@ func (o *projectOptions) WithServices(fn ProjectServicesFunc) func(cmd *cobra.Co
 			return err
 		}
 
-		if o.EnvFile != "" {
-			var services types.Services
-			for _, s := range project.Services {
-				ef := o.EnvFile
-				if ef != "" {
-					if !filepath.IsAbs(ef) {
-						ef = filepath.Join(project.WorkingDir, o.EnvFile)
-					}
-					if s.Labels == nil {
-						s.Labels = make(map[string]string)
-					}
-					s.Labels[api.EnvironmentFileLabel] = ef
-					services = append(services, s)
-				}
+		ef := o.EnvFile
+		if ef != "" {
+			if !filepath.IsAbs(ef) {
+				ef = filepath.Join(project.WorkingDir, o.EnvFile)
+			}
+		}
```

Comment:
```suggestion
		if ef != "" && !filepath.IsAbs(ef) {
			ef = filepath.Join(project.WorkingDir, ef)
		}
```
maybe?

Reply:
yes indeed, just a PoC to demonstrate the compose-go part of this  (should have created this as a Draft)